tests/test_interface.py

```diff
@@ -399,6 +415,28 @@ def test_interface_delete_payment_with_invalid_id(self, mock_db):
         self.assertNotEqual(payment_after_deletion.id,invalid_id)
         #TODO - next sprint : do get, check count, should be same
 
+#valid PUT data
+    @mock.patch.object(PaymentService,'is_valid_put',return_value = True)
+    @mock.patch.object(app_db, 'session')
+    @mock.patch('app.db.models.Payment')
+    #@mock.patch.object(Payment, 'deserialize_put')
+    #@mock.patch.object(Payment, 'serialize', return_value=PUT_CREDIT_RETURN)
+    def test_interface_update_with_valid_put_data(self,mock_isvalid,mock_db,mock_P):
```

Comment:
The order of the mocks here is incorrect - `mock_isvalid` should be the third of the three mock parameters, not the first.  See Derek's comment below for a correct ordering.